web/packages/teleport/src/MFAContext/MFAContext.tsx

```diff
@@ -0,0 +1,51 @@
+import { createContext, PropsWithChildren, useCallback, useState } from 'react';
+
+import AuthnDialog from 'teleport/components/AuthnDialog';
+import { useMfa } from 'teleport/lib/useMfa';
+import auth, { MfaChallengeScope } from 'teleport/services/auth/auth';
+import { MfaChallengeResponse } from 'teleport/services/mfa';
+
+export interface MfaContextValue {
+  getAdminActionMfaResponse(reusable?: boolean): Promise<MfaChallengeResponse>;
+}
+
+export const MfaContext = createContext<MfaContextValue>(null);
+
+export const MfaContextProvider = ({ children }: PropsWithChildren) => {
+  const adminMfa = useMfa({});
+
+  const getAdminActionMfaResponse = useCallback(
+    async (reusable: boolean = false) => {
```

Comment:
I'd replace convert it to an object, to make the callsites more readable. Also a JSDoc for it would be helpful.
```suggestion
    async (options: { reusable?: boolean } = {}) => {
```